runtime/src/chpl-cache.c

```diff
@@ -3566,12 +3566,32 @@ static struct rdcache_s* cache_create(void);
 CHPL_TLS_DECL(struct rdcache_s*,cache_remote_data);
 static pthread_key_t pthread_cache_info_key; // stores struct rdcache_s*
 
+// ideally we would just make is_inited an atomic but
+// the runtime atomics support can't handle static initialization
+// so we use a pthread mutex for now
+static pthread_mutex_t is_inited_mutex = PTHREAD_MUTEX_INITIALIZER;
+static unsigned char is_inited = 0; // protected by is_inited_mutex
+
+static
+unsigned char get_is_inited(void) {
+  unsigned char ret = 0;
+
+  pthread_mutex_lock(&is_inited_mutex);
+  ret = is_inited;
+  pthread_mutex_unlock(&is_inited_mutex);
+
+  return ret;
+}
```

Comment:
I found it a little confusing that reads of this var are wrapped, but writes aren't. Can you either put the write in a wrapper or inline this read. I have a minor preference for inlining the read with something like:

```c
if( !cache && chpl_cache_enabled()) {
  pthread_mutex_lock(&is_inited_mutex);
  chpl_bool local_is_inited = is_inited;
  pthread_mutex_unlock(&is_inited_mutex);
  if (local_is_inited) {
  ...

```